src/targets.rs

```diff
@@ -225,6 +248,43 @@ impl Targets {
   }
 }
 
+#[derive(Debug)]
+pub(crate) struct TargetsWithSupportsScope {
```

Comment:
Should this just be part of `Targets`?

Reply:
Do you mean putting the fields in `Targets` and making it stateful?